plugins/volumeFilter_test.go

```diff
@@ -145,3 +146,229 @@ func TestMakeFilterVolume(t *testing.T) {
+			inputOp:            makeManageSellOffer("2.0", "100.0"),
+			wantOp:             nil,
+			wantTbbBase:        pointy.Float64(0.0),
+			wantTbbQuote:       pointy.Float64(0.0),
+		},
+		{
+			name:               "9. selling, quote units sell cap, keep selling quote, new amount, exact mode",
+			mode:               volumeFilterModeExact,
+			sellBaseCapInBase:  nil,
+			sellBaseCapInQuote: pointy.Float64(1.0),
+			otbBase:            pointy.Float64(0.0),
+			otbQuote:           pointy.Float64(0.0),
+			tbbBase:            pointy.Float64(0.0),
+			tbbQuote:           pointy.Float64(0.0),
+			inputOp:            makeManageSellOffer("2.0", "100.0"),
+			wantOp:             makeManageSellOffer("2.0", "0.5000000"),
+			wantTbbBase:        pointy.Float64(0.5),
+			wantTbbQuote:       pointy.Float64(1.0),
+		},
+		{
+			name:               "10. selling, quote units sell cap, keep selling quote, new amount, ignore mode",
+			mode:               volumeFilterModeIgnore,
+			sellBaseCapInBase:  nil,
+			sellBaseCapInQuote: pointy.Float64(1.0),
+			otbBase:            pointy.Float64(0.0),
+			otbQuote:           pointy.Float64(0.0),
+			tbbBase:            pointy.Float64(0.0),
+			tbbQuote:           pointy.Float64(0.0),
+			inputOp:            makeManageSellOffer("2.0", "100.0"),
+			wantOp:             nil,
+			wantTbbBase:        pointy.Float64(0.0),
+			wantTbbQuote:       pointy.Float64(0.0),
+		},
+		{
+			name:               "11. selling, quote units sell cap, keep selling quote, exact mode",
+			mode:               volumeFilterModeExact,
+			sellBaseCapInBase:  nil,
+			sellBaseCapInQuote: pointy.Float64(1.0),
+			otbBase:            pointy.Float64(0.0),
+			otbQuote:           pointy.Float64(0.0),
+			tbbBase:            pointy.Float64(0.0),
+			tbbQuote:           pointy.Float64(0.0),
+			inputOp:            makeManageSellOffer("2.0", "100.0"),
+			wantOp:             makeManageSellOffer("2.0", "0.5000000"),
+			wantTbbBase:        pointy.Float64(0.5),
+			wantTbbQuote:       pointy.Float64(1.0),
+		},
+		{
+			name:               "12. selling, quote units sell cap, keep selling quote, ignore mode",
+			mode:               volumeFilterModeIgnore,
+			sellBaseCapInBase:  nil,
+			sellBaseCapInQuote: pointy.Float64(1.0),
+			otbBase:            pointy.Float64(0.0),
+			otbQuote:           pointy.Float64(0.0),
+			tbbBase:            pointy.Float64(0.0),
+			tbbQuote:           pointy.Float64(0.0),
+			inputOp:            makeManageSellOffer("2.0", "100.0"),
+			wantOp:             nil,
+			wantTbbBase:        pointy.Float64(0.0),
+			wantTbbQuote:       pointy.Float64(0.0),
+		},
+	}
+
+	// we fix the marketIDs and accountIDs, since volumeFilterFn output does not depend on them
+	marketIDs := []string{}
+	accountIDs := []string{}
+
+	for _, k := range testCases {
+		t.Run(k.name, func(t *testing.T) {
+			// exactly one of the two cap values must be set
+			if assert.Nil(t, k.sellBaseCapInBase) == assert.Nil(t, k.sellBaseCapInQuote) {
```

Comment:
when you run this, if one of them is non-nil then it will produce a test failure no? (because we are using assert to get the boolean value)

for a good way to achieve what we want without frivolous errors we can try this:
```golang
if k.sellBaseCapInBase == nil && k.sellbaseCapInQuote == nil {
    assert.Fail(t, "either one of the two cap values must be set")
    return
}
if k.sellBaseCapInBase != nil && k.sellbaseCapInQuote != nil {
    assert.Fail(t, "both of the cap values cannot be set")
    return
}
```

Reply:
Ah, this makes sense, thanks!